tests/cli/test_compose_tar_kickstart.sh

```diff
@@ -0,0 +1,158 @@
+
+        bp_name="test-tar"
+        blueprint="$bp_name.toml"
+        cat > $blueprint << __EOF__
+name = "$bp_name"
+description = "tar image test"
+version = "0.0.1"
+modules = []
+
+[[packages]]
+name = "kernel"
+version = "*"
+
+[[packages]]
+name = "openssh-server"
+version = "*"
+
+[[groups]]
+name = "anaconda-tools"
+
+[[customizations.user]]
+name = "root"
+key = "$pub_key"
+
+__EOF__
+        $CLI blueprints push $blueprint
+        image_path="/var/www/html/root.tar.xz"
+
+        version=$(awk -F = '$1 == "VERSION_ID" { print $2 }' /etc/os-release | tr -d \")
+        arch=$(uname -m)
+        baseurl=$(curl "https://mirrors.fedoraproject.org/mirrorlist?repo=fedora-${version}&arch=${arch}" | \
+            grep -v "^#" | head -n 1)
+        curl --remote-name-all $baseurl/images/pxeboot/{vmlinuz,initrd.img}
+
+        fallocate -l 5G disk.img
+    rlPhaseEnd
+
+    rlPhaseStartTest "compose start"
+        rlAssertEquals "SELinux operates in enforcing mode" "$(getenforce)" "Enforcing"
+        uuid=$($CLI compose start $bp_name tar)
+        rlAssertEquals "exit code should be zero" $? 0
+
+        uuid=$(echo $uuid | cut -f 2 -d' ')
+    rlPhaseEnd
+
+    rlPhaseStartTest "compose finished"
+        if [ -n "$uuid" ]; then
+            until $CLI compose info $uuid | grep 'FINISHED\|FAILED'; do
+                sleep 10
+                rlLogInfo "Waiting for compose to finish ..."
+            done;
+        else
+            rlFail "Compose uuid is empty!"
+        fi
+
+        rlRun -t -c "$CLI compose image $uuid"
+        image="$uuid-root.tar.xz"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Install tar image using kickstart liveimg command"
+        cat > $ks_path << __EOF__
+lang en_US.UTF-8
+timezone America/New_York
+keyboard us
+rootpw --lock
+sshkey --username root "$pub_key"
+bootloader --location=mbr
+zerombr
+clearpart --initlabel --all
+autopart
+# reboot is used together with --no-reboot qemu-kvm parameter for the qemu-kvm
+# process to exit after the installation is complete
+reboot
+
+liveimg --url http://10.0.2.2/root.tar.xz
+
+%post --log=/dev/console
+echo -e "curl http://10.0.2.2/image-$uuid-booted\n" >> /etc/rc.d/rc.local
+chmod +x /etc/rc.d/rc.local
+curl http://10.0.2.2/image-$uuid-installed
+%end
+__EOF__
+        mv $image $image_path
+        restorecon $image_path
+        $QEMU_BIN -m 2048 -drive file=disk.img,format=raw -nographic -kernel vmlinuz -initrd initrd.img \
+            -append "inst.ks=http://10.0.2.2/ks-tar.cfg inst.stage2=$baseurl console=ttyS0" --no-reboot &
+
+        timeout=60 # wait up to 60 minutes
+        t=0
+        while ((t < timeout)) && ! grep -q "image-$uuid-installed" /var/log/httpd/access_log; do
+            rlLogInfo "Waiting for the tar image to be installed."
+            sleep 1m
+            ((t += 1))
+        done
+        rlRun -t -c "grep image-$uuid-installed /var/log/httpd/access_log" 0 "Installation of the image finished."
```

Comment:
You can also add a rlLogInfo inside the if condition below to avoid grepping twice for the same thing.

Reply:
Good idea, thanks.

Reply:
I believe this is no more relevant, WRT changes made to how the VM is run.